fortitude/src/check.rs

```diff
@@ -905,17 +911,22 @@ pub fn check(args: CheckArgs, global_options: &GlobalConfigArgs) -> Result<ExitC
         .par_iter()
         .progress_with_style(progress_bar_style)
         .with_prefix("Checking file:")
-        .map(|path| {
+        .filter_map(|path| {
             let filename = path.to_string_lossy();
 
             let source = match read_to_string(path) {
                 Ok(source) => source,
                 Err(error) => {
-                    let message = format!("Error opening file: {error}");
-                    return Diagnostics::new(vec![DiagnosticMessage::from_error(
-                        filename,
-                        Diagnostic::new(IoError { message }, TextRange::default()),
-                    )]);
+                    if rules.enabled(Rule::IoError) {
```

Comment:
I'm not sure if we should allow `IoError` to be switched off. Doing so would also break the code in `write_summary_text` in `printer.rs`, which counts the number of `IoError` raised to determine the number of files which were scanned.

Reply:
Works fine for me:

```console
$ cargo run --bin fortitude -- check nope.f90 --ignore=E
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.28s
     Running `target/debug/fortitude check nope.f90 --ignore=E`
fortitude: 1 files scanned.
All checks passed!

```

The files we can't open get reported as files scanned, but this is probably fine.

Ruff issues a warning if you ignore `E902`:

```console
$ ruff check nope.py --ignore=E
warning: Failed to lint nope.py: No such file or directory (os error 2)
```

I've not done this yet as we probably want a proper logging system in place first. We could just write to stderr for now instead though.

Reply:
I would argue that it isn't working fine in that case, as it's still giving misleading output. A user could quite easily switch off the error reporting by accident, and end up being told their file has no errors when in fact it hasn't been scanned at all:

```bash
cargo run --bin fortitude check --select=typing typo.f90
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.08s
     Running `target/debug/fortitude check --select=typing typo.f90`
fortitude: 1 files scanned.
All checks passed!
```

If you would rather not fix it here, we could merge for now and raise a new issue for it.

Reply:
Yes, that's fair. I'll print a warning to stderr for now, but we should switch to a proper logger